tests/v1/kv_connector/run_accuracy_test.sh

```diff
@@ -20,26 +24,86 @@ wait_for_server() {
     done" && return 0 || return 1
 }
 
-# Prefill instance.
-CUDA_VISIBLE_DEVICES=0 VLLM_NIXL_SIDE_CHANNEL_PORT=5559 vllm serve $MODEL_NAME \
-    --port 8100 \
+# Arrays to store all hosts and ports
+PREFILL_HOSTS=()
+PREFILL_PORTS=()
+DECODE_HOSTS=()
+DECODE_PORTS=()
+
+# Start prefill instances
+for i in $(seq 0 $((NUM_PREFILL_INSTANCES-1))); do
+  # Calculate GPU ID - we'll distribute across available GPUs
+  GPU_ID=$((i % $(nvidia-smi --query-gpu=name --format=csv,noheader | wc -l)))
+  # Calculate port number (base port + instance number)
+  PORT=$((8100 + i))
+  # Calculate side channel port
+  SIDE_CHANNEL_PORT=$((5559 + i))
+
+  echo "Starting prefill instance $i on GPU $GPU_ID, port $PORT"
+
+  CUDA_VISIBLE_DEVICES=$GPU_ID VLLM_NIXL_SIDE_CHANNEL_PORT=$SIDE_CHANNEL_PORT vllm serve $MODEL_NAME \
+    --port $PORT \
     --enforce-eager \
     --disable-log-requests \
+    --gpu-memory-utilization 0.2 \
     --kv-transfer-config '{"kv_connector":"NixlConnector","kv_role":"kv_both"}' &
 
-# Decode instance.
-CUDA_VISIBLE_DEVICES=1 VLLM_NIXL_SIDE_CHANNEL_PORT=5558 vllm serve $MODEL_NAME \
-    --port 8200 \
+  # Store host and port for proxy configuration
+  PREFILL_HOSTS+=("localhost")
+  PREFILL_PORTS+=($PORT)
+done
+
+# Start decode instances
+for i in $(seq 0 $((NUM_DECODE_INSTANCES-1))); do
+  # Calculate GPU ID - we'll distribute across available GPUs, starting from after prefill GPUs
+  GPU_ID=$(((i + NUM_PREFILL_INSTANCES) % $(nvidia-smi --query-gpu=name --format=csv,noheader | wc -l)))
+  # Calculate port number (base port + instance number)
+  PORT=$((8200 + i))
+  # Calculate side channel port
+  SIDE_CHANNEL_PORT=$((5659 + i))
+
+  echo "Starting decode instance $i on GPU $GPU_ID, port $PORT"
+
+  CUDA_VISIBLE_DEVICES=$GPU_ID VLLM_NIXL_SIDE_CHANNEL_PORT=$SIDE_CHANNEL_PORT vllm serve $MODEL_NAME \
+    --port $PORT \
     --enforce-eager \
     --disable-log-requests \
+    --gpu-memory-utilization 0.2 \
```

Comment:
did you test on different GPUs for each instance? I am getting kv "oom" when sharing GPUs, although there should be plenty of space.